metricbeat/helper/server/http/http_test.go

```diff
@@ -42,18 +46,117 @@ func GetHttpServer(host string, port int) (server.Server, error) {
+2GA888CAT+birIE6EAwIyq7ZGe77ymRspugyb7AK46QOKApED3izxId36Tk5/a0P
+QY3WOTC0Y4yvz++gbx/uviYDMoHuJl0nIEXqtT9OZ2V2GqCToJu300RV/MIRtk6s
+0U1d9CRDkjNolGVbYo2VnDJbZ8LQtJHS5iDeiEztay5Cky4NvVZsbCxrgNrr3h/v
+upHEJ28Q7QzMnRC7d/THI6fRW1mG6BuFT3WPW5K7EAfgQDlyyspTDrACrYTuWC+y
+013uTlI=
+-----END CERTIFICATE-----`)
+
+		keyPem := []byte(`-----BEGIN RSA PRIVATE KEY-----
+MIIEowIBAAKCAQEAs4nZ3GkGxC1G+zaFK5SEOx/jDobezAWq0hct50+/PWs26h+b
+vkZ0D2mpeIjeEPDZm4glki1cTcV9VfGgT+6CwXDpxUsquelDx7Ei/wrQk25HbZ9o
+VxEuYPH9gvDTCY6czaZoFJp62szkXgLChWoOY7Bkez19sAOq2EAetsIDM1OSqfHY
+Hj9FZpg9tFaVJDsKtZAT1w0n9Yx1SPOd3kfQiJXuT9wcohteLf4k2KcJYcAAJZZL
+eKAOOMJl32UeQMfqpKk0axBTnLR6IUCKIFU+oMh1V89F7ikLzXPcOnpaqzS4dcdg
+eoGjdhaReeBMuUwtm0gwX6laUIlmSxyAuMv83wIDAQABAoIBAD1kY/T0jPXELcN1
+LzBpxpWZH8E16TWGspTIjE/Oeyx7XvnL+SulV8Z1cRfgZV8RnLeMZJyJmkiVwXgD
++bebbWbMP4PRYjjURPMh5T+k6RGg4hfgLIOpQlywIuoFg4R/GatQvcJd2Ki861Ii
+S3XngCgihxmFO1dWybLMqjQAP6vq01sbctUXYddFd5STInzrceoXwkLjp3gTR1et
+FG+Anmzbxp8e2ETXvwuf7eZhVwCJ2DxBt7tx1j5Csuj1LjaVTe5qR7B1oM7/vo0b
+LlY9IixAAi62Rrv4YSvMAtMI6mQt+AM/4uBVqoG/ipgkuoQVuQ+M4lGdmEXwEEkz
+Ol7SlMECgYEA11tV+ZekVsujBmasTU7TfWtcYtRHh+FSC040bVLiE6XZbuVJ4sSA
+TvuUDs+3XM8blnkfVo826WY4+bKkj1PdCFsmG5pm+wnSTPFKWsCtsSyA3ts85t3O
+IvcCxXA/1xL9O/UdWfrl2+IJ3yLDEjEU5QTYP34+KDBZM3u6tJzjWe8CgYEA1WwA
+8d75h9UQyFXWEOiwJmR6yX7PGkpYE3J7m2p2giEbLm+9no5CEmE9T74k3m0eLZug
+g/F1MA/evhXEYho6f+lS9Q0ZdtyU2EFrdvuLlUw6FJIWnaOLlVR/aC6BvAlxLDRb
+RUGqDKDjl1Die0s8F1aDHGvNvGaZRN4Z23BRPBECgYBE8pMGA8yzlSKui/SiE5iW
+UOcVJQ15rWPNBs62KZED5VdFr9cF6Q+DOfxe+ZWk+xHEDSdBWTylYPrgxpb05E6h
+vDzpHXfW64AO7jl18LYrQSpJLzvCVkUG4LpcZ+GohAXbSlCJXFB3I1kxvTli+5/K
+6tApE8vmpgQI/ZX6+Te4tQKBgBcQ3C1H5voaOf0c4czkCR2tIGQkk2eI/2nipp9O
+a053G4PySbEYOOXZopG6wCtV6bwOJNP9xaeTH4S1v4rGwOnQIsofR1BEWMXilCXA
+2/4fxesxOsaAxXY3Mqnk1NqovpWDdxXOGf3RaaeR81hV8kGndPYeZJbnE8uQoYTI
+586xAoGBAI2SR17xbgfiQBZxgGqamslz4NqBkZUBs4DIAGMAXS21rW/2bbbRaSii
+mGmkdaXx+l077AuO0peX2uBvJAx6PvAVW0qroeOLcCo6EuUGTNVhBej6L9hMwhIO
+r0tZLlMt75zcnJBicMbIrrzIGVYMHjT+m1QTGbrGb/tcEIGtmXwO
+-----END RSA PRIVATE KEY-----`)
+		cfg := &tls.Config{
+			ServerName: host,
+			MinVersion: tls.VersionTLS12,
+			MaxVersion: tls.VersionTLS12,
+			//Certificates:       c.Certificates,
+			//	RootCAs:            c.RootCAs,
+			//ClientCAs:          c.ClientCAs,
+			InsecureSkipVerify: true,
+			//CipherSuites:       c.CipherSuites,
+			//CurvePreferences:   c.CurvePreferences,
+			//ClientAuth:         c.ClientAuth,
+		}
+		cfg.Certificates = make([]tls.Certificate, 1)
+		cert, err := tls.X509KeyPair(certPem, keyPem)
+		if err != nil {
+			log.Fatal(err)
+		}
+		cfg.Certificates = []tls.Certificate{cert}
+		httpServer.TLSConfig = cfg
+	}
+
 	h.server = httpServer
 
 	return h, nil
 }
 
+func TestHttpsServer(t *testing.T) {
+	host := "127.0.0.1"
+	port := 40050
+	svc, err := GetHttpServer(host, port, true)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+
+	svc.Start()
+	defer svc.Stop()
+	// make sure server is up before writing data into it.
+	time.Sleep(2 * time.Second)
```

Comment:
It looks like we were using this sleep at the HTTP test. Doesn't looks solid.

Since we were already doing this let's go ahead, feel free to improve this dialing using tcp if you will.